src/udev/udev-rules-util.h

```diff
@@ -1,29 +1,58 @@
 #pragma once
 
+#include "escape.h"
 #include "macro.h"
+#include "utf8.h"
 
 static inline int parse_value(char *str, char **ret_value, char **ret_endpos) {
         char *value = str;
         char *i, *j;
+        int r;
+        bool is_escaped;
 
         /* value must be double quotated */
-        if (value[0] != '"')
+        if (value[0] == '"') {
+                is_escaped = false;
+                value++;
+        } else if (value[0] == 'e') {
+                if (value[1] != '"')
+                        return -EINVAL;
+                is_escaped = true;
+                value += 2;
+        } else {
                 return -EINVAL;
-        value++;
+        }
 
         /* unescape double quotation '\"' -> '"' */
         for (i = j = value;; i++, j++) {
                 if (*i == '"')
                         break;
                 if (*i == '\0')
                         return -EINVAL;
-                if (i[0] == '\\' && i[1] == '"')
+                if (i[0] == '\\' && i[1] == '"') {
+                        if (is_escaped) {
+                                *j = '\\';
+                                j++;
+                        }
                         i++;
+                }
                 *j = *i;
         }
         j[0] = '\0';
 
-        *ret_value = value;
+        if (is_escaped) {
+                char *unescaped;
+                r = cunescape_length(value, j - value, 0, &unescaped);
+                if (r < 0)
+                        return r;
+                assert(r < j - str);
+                memcpy(str, unescaped, r + 1);
+                free(unescaped);
+                *ret_value = str;
+        } else {
+                *ret_value = value;
+        }
```

Comment:
doesn't matter much, but we generally don#t put {} around single-line if blocks

Reply:
Just to double confirm. The if block has multiple lines but is broken by a earlier review comment.
Do we do this?
```
if (cond) {
        stmt1;
        stmt2;
} else
        stmt3;
```

Reply:
Exactly!